Single trial visualisation

charlotte-lemouel opened the pull request:
In the test.py file I added a function to visualise the results of the simulation

sheim commented:
Nice visualizations, but there are a few issues:
- Please replace {'Upwards', 'Fowards'} with either {'Vertical', 'Horizontal'} or {'x', 'y'}
- The foot appears to travel during stance in "Foot trajectory", which is just very confusing
- Not sure foot coordinates need to be plotted at all...

- The plotting function shouldn't be defined in the test.py script. This is just a demo script, and cannot be imported. This should either go into its own package (maybe utilities?) or in the same file as the model (this makes sense if the plotting is specific to the model). I understand you don't want to maintain the same plotting code for different models, then we just have to make sure the code remains generic enough (which shouldn't be a problem). So... `utilities.py` or `plotters.py` or something
- the plotting should be modularized: most of the time we won't want to plot foot stuff. Different plots should be split into separate functions, instead of having one call do all of them. Then you can also define a wrapper function which calls all your individual functions, and puts them into subfigures
- It would be nice to have different colors for stance and flight phases :)

charlotte-lemouel commented:
I moved the functions for plotting to a new folder called plotting, which has two files:

    single_trials now has 2 functions:
    full_visualisation: same as previous (with the changes in colors and labels that you asked for); I think it is useful for sanity checking to have one function which plots all the variables returned by the simulation. However, by default it no longer returns the foot trajectory.
    com_visualisation: nice plot of the CoM trajectory with the leg indicated, which we can use for graphs; if you like it, I will animate this.
    visualise_viability: for plotting the viability kernel

I've written examples of how to use these in the demos:

    test.py is modified to call single_trials
    I've added viability_plot.py which calls visualise_viability with your example data
